### PR TITLE
conductor: v1alpha2 integration

### LINKED ISSUE #205: conductor: update execution_client to support execution_api v1alpha2

### LINKED ISSUE #207: conductor: update executor to utilize execution api v1alpha2

### PR DESCRIPTION
**Pull request by joroshiba**
## Summary
Implements the v1alpha2 api, post refactor w/ updated tests.

## Background
Mostly the same as #400, but conductor refactored since then so updated around the refactor + test updates.

## Changes
- List changes which were made.

## Testing
Included tests and manual testing in dev-cluster against [astriaorg/go-ethereum#](https://github.com/astriaorg/go-ethereum/pull/29)

## Breaking Changelist
- Removes v1alpha1 API, requires new release of geth.

## Related Issues
closes #205
closes #207

### CODE REVIEW

File: crates/astria-conductor/src/execution_client.rs [hunk 1]
```diff
@@ -1,88 +1,154 @@
...
+        &mut self,
+        firm: Block,
+        soft: Block,
+    ) -> Result<ExecutorCommitmentState>;
 }
 
-impl ExecutionRpcClient {
-    /// Creates a new RPC Client
+#[async_trait::async_trait]
+impl ExecutionClientExt for ExecutionServiceClient<Channel> {
```
Comment (noot) [inline crates/astria-conductor/src/execution_client.rs:50]: where/how does the `<Channel>` generic get used? don't see it used anywhere, is it part of the autogenerated code?

File: crates/astria-conductor/src/executor/mod.rs [hunk 2]
```diff
@@ -161,12 +166,30 @@ impl Executor {
...
+                                            "failed to update soft commitment"
+                                        );
+                                    }
+                                }
+                                Ok(None) => {
+                                    error!(
+                                        height = height,
+                                        "block execution with empty return"
+                                    );
+                                }
```
Comment (noot) [inline crates/astria-conductor/src/executor/mod.rs:185]: this isn't an error, if empty block execution is disabled this is fine (and there's already a debug log for that case) so maybe just ignore this case?

File: crates/astria-conductor/src/executor/mod.rs [hunk 3]
```diff
@@ -254,14 +317,19 @@ impl Executor {
...
             .get(&sequencer_block_hash)
             .cloned();
-        match maybe_execution_block_hash {
-            Some(execution_block_hash) => {
-                self.finalize_block(execution_block_hash, sequencer_block_hash)
-                    .await?;
+        match maybe_executed_block {
+            Some(executed_block) => {
+                // this case means block has already been executed.
+                self.update_firm_commitment(executed_block.clone())
```
Comment (noot) [inline crates/astria-conductor/src/executor/mod.rs:327]: then you can remove the `clone()` here

File: crates/astria-conductor/src/executor/mod.rs [hunk 4]
```diff
@@ -254,14 +317,19 @@ impl Executor {
...
+                    .await
+                    .wrap_err("executor failed to update firm commitment")?;
+                // remove the sequencer block hash from the map, as it's been firmly committed
+                self.sequencer_hash_to_execution_block
+                    .remove(&block.block_hash);
```
Comment (noot) [inline crates/astria-conductor/src/executor/mod.rs:332]: and also can remove this

File: crates/astria-conductor/src/types.rs [hunk 5]
```diff
@@ -40,3 +44,31 @@ impl SequencerBlockSubset {
         }
     }
 }
+
+/// `ExecutorCommitmentState` is a struct that contains the firm and soft Blocks from the execution
```
Comment (noot) [inline crates/astria-conductor/src/types.rs:48]:
```suggestion
/// `ExecutorCommitmentState` is a struct that contains the firm and soft [`Block`]s from the execution
```

**Review by noot — APPROVED** [review]
looks good, minor comments only

File: crates/astria-conductor/src/execution_client.rs (hunk 1 above)
Comment (joroshiba) [reply crates/astria-conductor/src/execution_client.rs:50]: @steezeburger can you comment here, this was copied from previous PR

File: crates/astria-conductor/src/executor/mod.rs [hunk 6]
```diff
@@ -254,14 +317,19 @@ impl Executor {
             return Ok(());
         };
         let sequencer_block_hash = block.block_hash;
-        let maybe_execution_block_hash = self
-            .sequencer_hash_to_execution_hash
+        let maybe_executed_block = self
+            .sequencer_hash_to_execution_block
             .get(&sequencer_block_hash)
             .cloned();
```
[the remark this answers is not in the document]
Comment (joroshiba) [reply crates/astria-conductor/src/executor/mod.rs:323]: We don't want to remove unless commitment update is successful though.

File: crates/astria-conductor/src/execution_client.rs (hunk 1 above)
Comment (sambukowski) [reply crates/astria-conductor/src/execution_client.rs:50]: yeah this is part of the proto generated stuff

File: crates/astria-conductor/src/executor/mod.rs (hunk 2 above)
Comment (sambukowski) [reply crates/astria-conductor/src/executor/mod.rs]: you are correct! i'll remove the error! macro

File: crates/astria-conductor/src/executor/mod.rs (hunk 6 above)
[the remark this answers is not in the document]
Comment (sambukowski) [reply crates/astria-conductor/src/executor/mod.rs:319]: I think we need to keep this the same, the block gets removed from the map a few lines below after it has been finalized and I think we should keep that flow the same.
we don't want to remove the block if finalization fails for some reason.

File: crates/astria-conductor/src/executor/mod.rs (hunk 3 above)
Comment (sambukowski) [reply crates/astria-conductor/src/executor/mod.rs:323]: see comment above

File: crates/astria-conductor/src/executor/mod.rs (hunk 4 above)
Comment (sambukowski) [reply crates/astria-conductor/src/executor/mod.rs:328]: see comment above

**Review by sambukowski — APPROVED** [review]

**Pull request merged**